configs/machines/aleph.yaml

```diff
@@ -46,53 +46,127 @@ hyper_flag: ""
 #
 #
 
-useMPI: cray_mpich
-
-fc: ftn
-f77: ftn
-mpifc: ftn
-mpicc: cc
-cc: cc
-cxx: CC
-
-module_actions:
-        - "unload craype"
-        - "load craype/2.6.2"
-        - "load PrgEnv-cray/6.0.4"
-        - "load pbs"
-        - "load cray-mpich/7.7.3"
-        - "load craype-x86-skylake"
-        - "load cmake/3.14.0"
-        - "load cray-hdf5-parallel/1.10.2.0"
-        - "load cray-netcdf-hdf5parallel/4.6.1.3"
-        - "load cdo/1.9.5"
-        - "load fftw/2.1.5.9"
-        - "load nco/4.9.4"
-        - "load proj4/5.1.0"
-        - "load python/3.9.1"
-        - "list"
-
+useMPI: cray_mpich #intel_mpich #intelopenmpi #cray_mpich
+
+choose_useMPI:
+        cray_mpich:
+                fc: ftn
+                f77: ftn
+                mpifc: ftn
+                mpicc: cc
+                cc: cc
+                cxx: CC
+                module_actions:
+                        - "unload craype"
+                        - "load craype/2.6.2"
+                        - "load PrgEnv-cray/6.0.4"
+                        - "load pbs"
+                        - "load cray-mpich/7.7.3"
+                        - "load craype-x86-skylake"
+                        - "load cmake/3.14.0"
+                        - "load cray-hdf5-parallel/1.10.2.0"
+                        - "load cray-netcdf-hdf5parallel/4.6.1.3"
+                        - "load cdo/1.9.5"
+                        - "load fftw/2.1.5.9"
+                        - "load nco/4.9.4"
+                        - "load proj4/5.1.0"
+                        - "load python/3.9.1"
+                        - "list"
```

Comment:
A lot of this can go outside in a general `module_actions`. Here you can place only the differences. It's a matter of taste, I find it easier to find what the difference are in between the options if in the choose blocks there are only differences.

If you want to give it a shot I recommend to use here `add_module_actions` and also variable definition and called by placeholders in the general `module_actions` like in the Albedo example:

https://github.com/esm-tools/esm_tools/blob/release/configs/machines/albedo.yaml#L127-L149

https://github.com/esm-tools/esm_tools/blob/release/configs/machines/albedo.yaml#L103-L125

Reply:
If I place the defaults first, and then unload the craype, that might not work. The order can matter for module commands. Thus I try to keep them together as much as I can.

Reply:
It is indeed possible to keep the order if you follow the examples I pasted in the comment above

Reply:
I have time to give it a shot, if you have time to test it. Let me know what you prefer, doing it yourself or that I do it.

Reply:
I might find some time tomorrow. Otherwise it would sit till after I'm back.

Reply:
I'm back and would have time to test.